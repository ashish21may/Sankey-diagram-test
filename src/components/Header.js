import React from 'react'
import '../styles/Header.css'
import { useTranslation } from 'react-i18next'

const Header = () => {
  const { i18n } = useTranslation()

  const handleOnClick = () => {
    if (i18n.language === 'en') {
      i18n.changeLanguage('hin')
    } else {
      i18n.changeLanguage('en')
    }
  }

  return (
    <div className='header'>
      <div data-testid='centime-header' className='header-text'>
        Centime
      </div>
      <button
        className='header-language-button'
        data-testid='language-button'
        onClick={handleOnClick}
      > {i18n.language}
      </button>
    </div>
  )
}

export default Header
