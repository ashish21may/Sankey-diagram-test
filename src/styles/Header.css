.header {
  display: flex;
  height: 50px;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  background-color: cornflowerblue;
  color: white;
  font-size: xx-large;
}

.header-text {
  margin-left: 45%;
}

.header-language-button {
  background-color: firebrick;
  font-size: medium;
  color: white;
  cursor: pointer;
  width: 75px;
  height: 32px;
  border-radius: 10%;
  margin-right: 5%;
}