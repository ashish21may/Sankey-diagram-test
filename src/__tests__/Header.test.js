import React from 'react'
import Header from '../components/Header'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: key => key,
    i18n: { changeLanguage: jest.fn() }
  })
}));

const onClick = jest.fn();

describe('Header', () => {
  it('Header snapshot', () => {
    // Fragment is html returned by Header component.
    const { asFragment } = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const { getByTestId, getByRole } = render(<Header />)
    expect(getByTestId('centime-header')).toHaveClass('header-text')
    expect(getByTestId('centime-header')).toHaveTextContent('Centime')
    expect(getByRole('button')).toBeInTheDocument()
  })
})
