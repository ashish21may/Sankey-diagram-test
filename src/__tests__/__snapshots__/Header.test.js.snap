// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header-text"
      data-testid="centime-header"
    >
      Centime
    </div>
    <button
      class="header-language-button"
      data-testid="language-button"
    >
       
    </button>
  </div>
</DocumentFragment>
`;
